src/utils/components/SSHSecretSection/utils/components/SelectSecret.tsx

```diff
@@ -0,0 +1,86 @@
+import * as React from 'react';
+
+import { SecretModel } from '@kubevirt-ui/kubevirt-api/console';
+import { IoK8sApiCoreV1Secret } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import { validateSSHPublicKey } from '@kubevirt-utils/utils/utils';
+import { ResourceLink, WatchK8sResult } from '@openshift-console/dynamic-plugin-sdk';
+import { Alert, AlertVariant, Select, SelectOption, SelectVariant } from '@patternfly/react-core';
+
+import Loading from '../../../Loading/Loading';
+import { decodeSecret } from '../utils';
+
+type SelectSecretProps = {
+  secretsResourceData: WatchK8sResult<IoK8sApiCoreV1Secret[]>;
+  selectedSecretName: string;
+  onSelectSecret: (secretName: string) => void;
+  id?: string;
+};
+
+const SelectSecret: React.FC<SelectSecretProps> = ({
+  secretsResourceData,
+  selectedSecretName,
+  onSelectSecret,
+  id,
+}) => {
+  const { t } = useKubevirtTranslation();
+  const [isSecretSelectOpen, setSecretSelectOpen] = React.useState(false);
+
+  const [secrets, secretsLoaded, secretsError] = secretsResourceData;
+  const sshKeySecrets = secrets
+    ? secrets?.filter((secret) => validateSSHPublicKey(decodeSecret(secret)))
+    : [];
+
+  const filterSecrets = (_, value: string): React.ReactElement[] => {
+    let filteredSecrets = sshKeySecrets;
+
+    if (value) {
```

Comment:
What happen if you remove this if statement and set default value to value of "" ?

Reply:
What default value?

Ugo wrote this. I just copied it over to this location since the other SSH code will eventually go away. That allowed me to do a little cleanup. I'm not sure what this regex is supposed to be doing, though. @upalatucci Can you explain what this is doing?

Additionally, this is why I think we should avoid using `value` as a parameter name when a more descriptive name can be used. I'm guilty of doing it myself, but we should start holding each other accountable for it, IMO.

Reply:
The `value` here is just what the user type in the searchbox when the dropdown is open. If the value is undefined or empty, the user should see all the secrets. If the user type something, should see the options filtered.

Sorry for this sad naming. You are right about that.

Reply:
Thanks, Ugo! That makes sense. I'll update the name. No need to apologize. We're all guilty of doing it from time to time.